src/main/java/gov/cabinetoffice/gap/adminbackend/services/SchemeService.java

```diff
@@ -66,6 +70,16 @@ public Integer postNewScheme(SchemePostDTO newScheme, HttpSession session) {
             if (featureFlagsConfigurationProperties.isNewMandatoryQuestionsEnabled()) {
                 entity.setVersion(2);
             }
+
+            Optional<GrantAdmin> grantAdmin = this.grantAdminRepository.findById(adminSession.getGrantAdminId());
```

Comment:
could move this into a nice wee functional block 

```
            this.grantAdminRepository.findById(adminSession.getGrantAdminId())
                    .ifPresentOrElse(
                            entity::addAdmin, 
                            () -> new SchemeEntityException("Something went wrong while creating a new grant scheme: No grant admin found for id: " + adminSession.getGrantAdminId()));

Reply:
nice!

Reply:
actually it would need to be this 

```
            this.grantAdminRepository.findById(adminSession.getGrantAdminId())
                    .ifPresentOrElse(
                            entity::addAdmin,
                            () -> {
                                throw new SchemeEntityException("Something went wrong while creating a new grant scheme: No grant admin found for id: " + adminSession.getGrantAdminId());
                            }
                    );